src/auth.ts

```diff
@@ -25,4 +28,28 @@ async function signOut({ returnTo }: { returnTo?: string } = {}) {
   await terminateSession({ returnTo });
 }
 
-export { getSignInUrl, getSignUpUrl, signOut };
+export async function switchToOrganization(
+  organizationId: string,
+  options: SwitchToOrganizationOptions = {},
+): Promise<UserInfo> {
+  const { returnTo, revalidationStrategy = 'path', revalidationTags = [] } = options;
+  const headersList = await headers();
+  const pathname = returnTo || headersList.get('x-url') || '/';
+  const result = await refreshSession({ organizationId, ensureSignedIn: true });
+
+  switch (revalidationStrategy) {
+    case 'path':
+      revalidatePath(pathname);
+      break;
+    case 'tag':
+      for (const tag of revalidationTags) {
+        revalidateTag(tag);
+      }
+      break;
+  }
+  if (revalidationStrategy !== 'none') {
```

Comment:
I'd have to test this piece to ensure the other creations of auth tokens with widgets don't need both (not that the developer can't add this to their wrapper function).